lib/ansible/modules/system/macos_pref.py

```diff
@@ -0,0 +1,1079 @@
+    - Dates need to be quoted.
+    - First level (not nested) quoted boolean, integers and floats are
+      converted to boolean, integer and float unless C(type) is specified.
+    - Lists and dicts can't be given as C(value) in C(key=argument) form.
+    - Binary data needs to be encoded in base64.
+'''
+
+EXAMPLES = '''
+# Set a string key
+- name: Set preferred view style
+  macos_pref:
+    domain: com.apple.finder
+    key: FXPreferredViewStyle
+    value: Nlsv
+
+
+# Read a string key
+- name: Read preferred view style
+  macos_pref:
+    domain: com.apple.finder
+    key: FXPreferredViewStyle
+  register: finder_view_style
+
+- name: Check preferred view style
+  fail: msg="Only list view is acceptable!"
+  when: finder_view_style.value != 'Nlnv'
+
+
+# Set a boolean key
+- name: Show all files
+  macos_pref:
+    domain: com.apple.finder
+    key: AppleShowAllFiles
+    value: true
+
+
+# Read a boolean key
+- name: Show all files
+  macos_pref:
+    domain: com.apple.finder
+    key: AppleShowAllFiles
+  register: show_all_files
+
+- name: Check show all files
+  fail: msg="Only show all files is acceptable!"
+  when: not show_all_files.value
+
+
+# Set (replace) a nested dictionary key
+- name: Set desktop and finder icon view settings
+  macos_pref:
+    domain: com.apple.finder
+    key: FK_StandardViewSettings:IconViewSettings
+    value:
+      arrangeBy: name
+      gridSpacing: 44
+      iconSize: 36
+      showIconPreview: true
+      showItemInfo: false
+      labelOnBottom: true
+      textSize: 12
+
+
+# Set (merge) a dictionary key
+- name: Set desktop and finder icon size
+  macos_pref:
+    domain: com.apple.finder
+    key: FK_StandardViewSettings
+    value:
+      IconViewSettings:
+        iconSize: 50
+    state: merge
+
+
+# Read a dictionary key
+- name: Show all files
+  macos_pref:
+    domain: com.apple.finder
+    key: FK_StandardViewSettings
+  register: finder_view_settings
+
+- name: Check show all files
+  fail: msg="Only iconSize=50 is acceptable!"
+  when: finder_view_settings.value.IconViewSettings.iconSize != 50
+
+
+# Delete a key
+- name: Set Terminal default window settings
+  macos_pref:
+    domain: com.apple.Terminal
+    key: Default Window Settings
+    state: absent
+
+
+# Read directly from a plist file (also works for writing)
+- name: Read rumour from file
+  macos_pref:
+    domain: /tmp/rumours.plist
+    key: Rumour
+  register: rumour
+
+
+# Set a dict on currentHost
+- name: Set screensaver
+  macos_pref:
+    domain: com.apple.screensaver
+    host: currentHost
+    key: moduleDict
+    value:
+      moduleName: Flurry
+      displayName: Flurry
+      path: /System/Library/Screen Savers/Flurry.saver
+      type: 0
+
+
+# Set the nested key arrangeBy in abbreviated form (nested keys need to be quoted).
+- macos_pref: { domain: com.apple.finder, key: 'DesktopViewSettings:IconViewSettings:arrangeBy', value: dateModified }
+'''
+
+RETURN = '''
+value:
```

Comment:
I wonder whether `data` is not a better name than value.
I think it's used more often by other modules.

Reply:
macOS preferences are key/value pairs, so shouldn't you get a `value` in return when you read a `key`? When setting a key you provide the value in `value` as well.